llvm/utils/TableGen/GlobalISelMatchTable.cpp

```diff
@@ -39,10 +40,46 @@ std::string getEnumNameForPredicate(const TreePredicateFn &Predicate) {
 std::string getMatchOpcodeForImmPredicate(const TreePredicateFn &Predicate) {
   return "GIM_Check" + Predicate.getImmTypeIdentifier().str() + "ImmPredicate";
 }
+
+// GIMT_Encode2/4/8
+constexpr StringLiteral EncodeMacroName = "GIMT_Encode";
+
 } // namespace
 
 //===- Helpers ------------------------------------------------------------===//
 
+void emitEncodingMacrosDef(raw_ostream &OS) {
```

Comment:
Why not just put these prints in `emitEncodingMacrosDef` and `emitEncodingMacrosUndef` to  `emitMatchTable`? So that we can see the whole structure of generated code.